Site Migration: Update migration instructions screen

gabrielcaires opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->



Related to #88588

## Proposed Changes
* Update the site migration instructions page to reflect the figma layout  5yhaC6umexdQLlLpCrCyEe-fi-2095_9011
* Update the instructions
* Automatically load the migration key without any user interaction
* Show a different message when we can’t get the migration link
* Add links directly to the related pages

More context: p1711029899474269-slack-C0Q664T29




## Testing Instructions
**SHORT VERSION:**
1. Enable the feature toggle
2. Access `/setup/site-migration/site-migration-instructions?from=[JN site]&siteSlug=[YOUR_WORDPRESS.COM site created by previous migrations]`
3. Jump to the steps 10, 11 and 12.


**LONG VERSION:**
1. Go to the `/start` URL using the calypso.live instructions
2. Follow the flow until see the "Goals" page
4. Open the browser Web developer tools and copy and paste the following code `sessionStorage.setItem('flags', "onboarding/new-migration-flow")`
5. Type Enter
6. Reload the goals page
7. Select "Import or migrate content"
8. "Where will you import from?" set `https://test-vanilla-wp.godaddy.test-wpcom-migrations.net/`
9. Select "Everything (requires a Creator Plan)" on "What do you want to migrate?"
10. When you arrive at the "Ready to migrate your site?" follow the instructions using the page links and copying the migration key. **Screenshot 1(a,b)**
11. After following the instructions, refresh the page to see a screen variation. **Screenshot 2(a,b)**
12. Check again the instructions for the new version, using the links.

NOTE: The screen variation with the migration key is only shown the first time the user sees the screen, after a reload only the second variation is shown. 

NOTE 2: If you want to reset the site state to see screen 1 again,  you just need to open the URL([TARGET_SITE]/_cli) and run the command `wp option delete wpcom_site_migration_migrate_guru_key_read` to delete the site option we are using to not show again the migration key.


### Screenshots
**1a. Desktop - Migration key available**
<img width="1022" alt="image" src="https://github.com/Automattic/wp-calypso/assets/38718/1d91384d-5c53-4793-877f-f6400b529691">

**1.b Mobile - Migration key available**
<img width="408" alt="image" src="https://github.com/Automattic/wp-calypso/assets/38718/b1d00d5f-600e-4bd0-9fcc-261fcdd4e0dd">

**2a. Desktop - Migration key not available**
<img width="1028" alt="image" src="https://github.com/Automattic/wp-calypso/assets/38718/018d2588-718d-4565-8988-ea082f3ea7b8">

**2b. Mobile - Migration key not available**
<img width="380" alt="image" src="https://github.com/Automattic/wp-calypso/assets/38718/0c700d1f-888c-4589-8a41-7e0f2c749387">


Interactions:

https://github.com/Automattic/wp-calypso/assets/38718/6082efbb-1069-4526-811b-aaba9e30b60c

gabrielcaires committed:
Add hook to get the site-migration-key

gabrielcaires committed:
Add component to show/hide the migration key

gabrielcaires committed:
Update layout

gabrielcaires committed:
Add animation

gabrielcaires commented:
@javierarce I added a video showing the animations and interactions but feel free to see the real screen.

gabrielcaires commented:
@javierarce The show/hide eyes icon is different from the layout because there is a recommendation to use the wordpress/icons library 

https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library

<img width="592" alt="image" src="https://github.com/Automattic/wp-calypso/assets/38718/c920824d-38ea-4fad-9f66-58840779db00">

javierarce commented:
> @javierarce The show/hide eyes icon is different from the layout because there is a recommendation to use the wordpress/icons library

You are totally right, sorry about that, I'll update the icon in Figma.

Regarding the implementation of the design, it looks great, but if we made two little adjustments to the heights of the button and the input field, they'd look better. Could we make the height of the input field 44px and the height of the button 28px?

![image](https://github.com/Automattic/wp-calypso/assets/4933/37939237-1b4a-41f6-938c-5209fa85cce8)

@michaelpick could you and Francisco review the copy of the instructions step (5yhaC6umexdQLlLpCrCyEe-fi-2095_9011)? I think I never showed them to you.

Here's the text:

> Ready to migrate your site?
> Follow these steps to get started.
>
> 1. Install and activate the Migrate Guru plugin on your existing site.
> 2. In the Migrate Guru plugin settings page, enter your email address and click Migrate.
> 3. When asked to select a destination host, pick “Automattic.”
> 4. Copy and paste the migration key below in the “Migrate Guru Migration Key” field and click Migrate.
> 
> And you are done! When the migration finishes, Migrate Guru will send you an email.

gabrielcaires committed:
Adjust migration key input size

gabrielcaires commented:
> Regarding the implementation of the design, it looks great, but if we made two little adjustments to the heights of the button and the input field, they'd look better. Could we make the height of the input field 44px and the height of the button 28px?
> 

![image](https://github.com/Automattic/wp-calypso/assets/38718/f12bad99-b39f-4c9f-a4f5-6f6b22d0e6c1)
![image](https://github.com/Automattic/wp-calypso/assets/38718/ca435d25-f60b-4f25-b69b-0ed704d0f22c)

@javierarce size updated! 

I also prevented the component from automatically selecting the key when it is hidden.

valterlorran commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx:
```diff
@@ -1,185 +1,129 @@
-							},
-						}
-					) }
+					{ translate( 'When asked to select a destination host, pick {{em}}“Automattic”{{/em}}.', {
```
Can we set the name of the host to "WordPress.com"? They already updated it.

gabrielcaires replied:
Updated here  cc012deec311e73a0a9e059bd637364e965f9f60

javierarce commented:
> @javierarce size updated!
> I also prevented the component from automatically selecting the key when it is hidden.

Fantastic, much better, thanks!

javierarce commented:
One last thing that I noted (but we can update this after we get the content review) is the use of italics. Since we are using quotes the italics are redundant.

daledupreez commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx:
```diff
@@ -1,185 +1,129 @@
 				<li>
 					{ translate(
-						'Go to the {{a}}Migrate Guru page on the source site{{/a}}, enter your email address, and click {{strong}}Migrate{{/strong}}.',
+						'Install and active the {{a}}Migrate Guru plugin{{/a}} on your existing site.',
```
Typo.
```suggestion
						'Install and activate the {{a}}Migrate Guru plugin{{/a}} on your existing site.',
```

gabrielcaires replied:
Typo fixed here https://github.com/Automattic/wp-calypso/pull/88813/commits/658ae3d6c4e12a09c2d425af37902f8c47574824

daledupreez commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx:
```diff
@@ -1,185 +1,129 @@
+				{ isSuccess && migrationKey && (
+					<li>
+						{ translate(
+							'Copy and paste the migration key below in the {{em}}“Migrate Guru Migration Key”{{/em}} field and click {{strong}}Migrate{{/strong}}.',
```
Question: are the label and button CTA translated in the Blog Vault/Migrate Guru UI?
We run the risk of translating these terms when they are not translated in the underlying screen.

gabrielcaires replied:
Good question!
I checked and saw they were not translating the CTAs, so I applied this change https://github.com/Automattic/wp-calypso/pull/88813/commits/d22bc4dfa381493da09fd3ac2b3f2df5b83a0202 to protect then to be translated.

daledupreez commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx:
```diff
@@ -1,185 +1,129 @@
 				<li>
 					{ translate(
-						'You will see a screen showing multiple hosting providers. Select {{em}}Automattic{{/em}} as the destination host.',
+						'Go to the {{a}}Migrate Guru page on the source site{{/a}}, enter your email address, and click {{strong}}Migrate{{/strong}}.',
```
Minor note (which is missing _all_ the context!): might it be clearer if we used "on ~the~ your source site"?

gabrielcaires replied:
Fixed here https://github.com/Automattic/wp-calypso/pull/88813/commits/658ae3d6c4e12a09c2d425af37902f8c47574824

daledupreez reviewed (commented):
I have some quick feedback from inspection.

gabrielcaires committed:
Replace Automattic to WordPress.com

gabrielcaires committed:
Remove quotes

gabrielcaires committed:
Add log when we are showing the migration key fallback instructions

Improve copy

gabrielcaires committed:
Improve copy

gabrielcaires commented:
> One last thing that I noted (but we can update this after we get the content review) is the use of italics. Since we are using quotes the italics are redundant.

Sure, quotes removed 9f9478c4695797b0feef80530d50deebdbc9c125

gabrielcaires committed:
Fix test

gabrielcaires committed:
Protect the CTA to be translated

Currently, the main Migrate Guru CTA are not translatable, so we want to protect the button and fields names to be translated from our translation team.

andres-blanco commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx:
```diff
@@ -1,185 +1,151 @@
+							{
+								components: {
+									migrationKeyField: (
+										<DoNotTranslateIt value="Migrate Guru Migration key" as="em" />
```
What's the reasoning behind adding a `DoNotTranslateIt` component? 

maybe adding the markup in the input string directly and passing the `Migrate Guru Migration key` | `Migrate` strings as an [arg](https://github.com/Automattic/wp-calypso/blob/trunk/packages/i18n-calypso/types/index.d.ts#L34) is enough?

gabrielcaires replied:
@andres-blanco 
The idea is to express the intent the value should not be translated, especially considering we don't have any kind of automated test testing translations.

My concern is we can easily remove the args value just because the intent is not clear and we don't have a test suite testing translations.

andres-blanco replied:
ok, then my only comment is that it would be good to remove the markup from the `DoNotTranslateIt` element and keep it in the input string. That way is more readable, and also extendable (imagine if you want to do something like `<strong><em>STRING</em></strong>`).
But it's just a nitpick, not a blocker. Go with your gut :)

gabrielcaires replied:
Good point; I tested, and we can still do it via `{{strong}} {{migrationKeyField /}} {{/ strong }}.` but I still agree with you because it blocks the translators to freely move the formatting position. 
So, I update here https://github.com/Automattic/wp-calypso/pull/88813/commits/17eefe3a9baba824878040c44fb9501043d416b1

gabrielcaires committed:
Fix typescript checking

gabrielcaires committed:
Fix typescript checking

gabrielcaires committed:
Remove tag from DoNotTranslateIt

gabrielcaires committed:
Add missing strong tag

gabrielcaires committed:
Fix unexpected scroll

daledupreez commented on client/landing/stepper/hooks/use-site-migraiton-key.ts:
```diff
@@ -0,0 +1,22 @@
+import { useQuery } from '@tanstack/react-query';
```
Nit: the filename has a typo. `-migraiton-` => `-migration-`

gabrielcaires replied:
Fixed https://github.com/Automattic/wp-calypso/pull/88952/commits/694f4fccc9098322d923d169e7b9f886fe9d15f3

daledupreez commented on client/landing/stepper/hooks/use-site-migraiton-key.ts:
```diff
@@ -0,0 +1,22 @@
+	} );
+
+export const useSiteMigrationKey = ( siteId?: number ) => {
+	return useQuery( {
+		queryKey: [ 'site-migration-key', siteId ],
+		queryFn: () => getMigrationKey( siteId! ),
+		retry: false,
+		enabled: !! siteId,
+		select: ( data ) => ( { migrationKey: data?.migration_key } ),
+		refetchOnWindowFocus: false,
+	} );
```
Should we specify `staleTime` on the `useQuery()` call to allow the data to be kept in memory for a little while before we try to refetch? ([reference](https://tanstack.com/query/v4/docs/framework/react/reference/useQuery))

gabrielcaires replied:
I set 60 seconds as staleTime as we discussed on slack. 
https://github.com/Automattic/wp-calypso/pull/88952/commits/a8022e45bee59130156286d12da9f4d8ffe1a473

daledupreez commented on client/landing/stepper/hooks/use-site-migraiton-key.ts:
```diff
@@ -0,0 +1,22 @@
+export const useSiteMigrationKey = ( siteId?: number ) => {
+	return useQuery( {
+		queryKey: [ 'site-migration-key', siteId ],
+		queryFn: () => getMigrationKey( siteId! ),
```
Is the `!` intentional? If so, what does it do?!?

gabrielcaires replied:
Yes, It is saying that typescript can assume the siteId will be available when the function is executed. 
The enabled flag is already ensuring it.

The line above ensures that react-query only triggers the queryFn when the siteId is available. 
``` 
enabled: !! siteId,
`` `

daledupreez commented on client/landing/stepper/hooks/test/use-site-migration-key.tsx:
```diff
@@ -0,0 +1,39 @@
+			.once()
+			.reply( 200, { migration_key: 'some-migration-key' } );
+
+		const { result } = renderHook( () => useSiteMigrationKey( 'some-site-id' ), { wrapper } );
```
Nit: the argument expects the siteId argument to be numeric, so we should likely test it with a number (or allow a site slug to be specified in the types).

gabrielcaires replied:
Fixed here https://github.com/Automattic/wp-calypso/pull/88952/commits/67c80a1df84213e1ab9cc8af355b9eba6d327ef9

We are with the typescript checking disabled on the test files on our pipeline😢 .

daledupreez commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/show-hide-input/style.scss:
```diff
@@ -0,0 +1,58 @@
+	max-width: 456px;
+	transition: all 0.2s;
+
+	&--hidden {
```
Nit: I have a strong personal preference for avoiding `&` as selector/text duplicator in CSS selectors because it makes it much harder to find the selector when debugging in an IDE.

While that may not seem like a common use case, every now and then we hit issues with CSS selectivity in Calypso as a result of duplicate selector names that are defined in different CSS bundles, and knowing where and how the rules were defined is relatively important.

Also, I find I need to think extra-hard to make sure I know what `&` actually represents, especially within patch diffs where the wrapping selector may not be visible.

gabrielcaires replied:
Thanks for the context:
I removed https://github.com/Automattic/wp-calypso/pull/88952/commits/b24b270f9e140f52eb4ea1f4dc624f7a8f1e3b21 the usage of `&`

daledupreez commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx:
```diff
@@ -1,185 +1,152 @@
 				<li>
 					{ translate(
-						'You will see a screen showing multiple hosting providers. Select {{em}}Automattic{{/em}} as the destination host.',
+						'Go to the {{a}}Migrate Guru page on your source site{{/a}}, enter your email address, and click {{strong}}{{migrateButton /}}{{/strong}}.',
 						{
 							components: {
-								em: <em />,
+								a: <a href={ getMigrateGuruPageURL( fromUrl ) } target="_blank" rel="noreferrer" />,
+								migrateButton: <DoNotTranslateIt value="Migrate" />,
+								strong: <strong />,
 							},
 						}
 					) }
```
I don't have a super-strong opinion on this, but I do think it would still be relatively easy and clear if we handle this with args and a comment. (We may need the comment anyway.) e.g.
```suggestion
					{ translate(
						'Go to the {{a}}Migrate Guru page on your source site{{/a}}, enter your email address, and click {{strong}}%(migrate)s{{/strong}}.',
						{
							comment: 'Note that the migrate arg will always be "Migrate", as the Blog Vault UI is not translated',
							args: {
								migrate: 'Migrate',
							},
							components: {
								a: <a href={ getMigrateGuruPageURL( fromUrl ) } target="_blank" rel="noreferrer" />,
								strong: <strong />,
							},
						}
					) }
```

gabrielcaires replied:
The feedback I received was that the translators sometimes don't read the comments. 

Regarding the args, I explained my concern here. 
https://github.com/Automattic/wp-calypso/pull/88813#discussion_r1539303167

daledupreez reviewed (approved):
Overall, this is LGTM, though I do have some non-blocking feedback.

javierarce commented:
> @javierarce Sure! I removed the quotes here https://github.com/Automattic/wp-calypso/commit/9f9478c4695797b0feef80530d50deebdbc9c125

Oh, I meant to remove the italics and leave the quotes. We don't italicize that type of text.

I've pinged Francisco to do a content review of the text. If possible, let's wait for it before we merge this PR.

javierarce commented:
Ok, here are the corrections (1e1T4v-g_4Lu6Kv2uCRIRYGr7KAlqlv0hxcpTm34FWbA-gdoc):

![image](https://github.com/Automattic/wp-calypso/assets/4933/1e156752-671f-4162-b9d0-acfb052fd0ce)

```
Title: Ready to migrate your site?
Subtitle: Follow these steps

1. Install and activate the [Migrate Guru plugin] on your existing site.
2. Go to the [Migrate Guru page on your source site], enter your email address, and click **Migrate**.
3. When asked to select a destination host, pick ‘WordPress.com‘
4. Copy and paste the migration key below in the ‘Migrate Guru Migration Key‘ field and click **Migrate** once more.

And you are done! When the migration is ready, Migrate Guru will send you an email.
```

![image](https://github.com/Automattic/wp-calypso/assets/4933/ce7e9fe2-94a2-4378-b00a-4be606744d2b)

```
Title: Ready to migrate your site?
Subtitle: Follow these steps

1. Install and activate the [Migrate Guru plugin] on your existing site.
2. Go to the [Migrate Guru page on your source site], enter your email address, and click Migrate.
3. When asked to select a destination host, pick ‘WordPress.com‘
4. Head to the [Migrate Guru page on the new WordPress.com site] and copy the migration key. 
5. Paste the migration key on the ‘Migrate Guru Migration Key’ field of your existing site and click **Migrate** once more.

And you are done! When the migration is ready, Migrate Guru will send you an email.
```

gabrielcaires commented:
@javierarce @daledupreez Due to some github bug I only could see your suggestions after I merged. 
I am going to open a new PR to fix your comment.
